src/Tinfoil/Random.hs

```diff
@@ -80,3 +82,10 @@ drawOnce as = do
     n = NE.length as
 
     discard = dropMS extraBits
+
+-- | Generate identifier unique to the currently-running kernel.
+--
+-- https://github.com/torvalds/linux/blob/22763c5cf3690a681551162c15d34d935308c8d7/drivers/char/random.c
+linuxBootId :: IO BootId
+linuxBootId =
+  fmap (BootId . T.encodeUtf8) $ T.readFile "/proc/sys/kernel/random/boot_id"
```

Comment:
I'm not sure if `/proc` can fail to mount in a real sense - it can be mounted in the wrong place or not mounted at all, but I think that's morally the same as if we were running on a non-linux system, we'd get an exception from `readFile` and die there. We won't be seeing ioerrors or anything like that as there's no physical storage underneath /proc; as long as the kernel is responding to syscalls /proc will be readable. (I did look into using an FFI call to the `sysctl` syscall, but although it's possible the manpage recommends against it in strong enough terms that I reconsidered). It's annoying that it's not portable, am definitely open to suggestions wrt possible alternatives for a machine identifier.